### ISSUE TITLE
Error: must implement interface TYPO3\CMS\Backend\View\PageLayoutViewDrawItemHookInterface

### ISSUE THREAD

**Issue by ghost** [NamelessCoder/inline_page_editing#1]
Installing EXT:inline_page_editing v. 1.0.1 (although version number in Extension Manager says 1.0.0!) on a TYPO3 installation with EXT:typo3-themes and EXT:news, I get this error when clicking the Page module (screenshot below):

> Oops, an error occurred!
> 
> NamelessCoder\InlinePageEditing\Hooks\ContentPreview must implement interface TYPO3\CMS\Backend\View\PageLayoutViewDrawItemHookInterface

![inline-page-editing-error-1](https://cloud.githubusercontent.com/assets/1712808/18185385/b544dc88-709d-11e6-8318-3696ae9c4892.png)

**Comment by NamelessCoder** [comment]
Needs `composer install` (this error is raised when the ContentPreview class from this extension cannot be loaded by the composer autoloader) - if you clone the repository into typo3conf/ext manually then you must add class loading information manually, for example in the root composer.json of the site:

```
{
    "require": {
        "typo3/cms": "^7.6",
        "namelesscoder/inline-page-editing": "^1.0"
    },
    "autoload": {
        "psr-4": {
            "NamelessCoder\\InlinePageEditing\\": "typo3conf/ext/inline_page_editing/Classes/"
        }
    }
}
```

(and then composer dump-autoload).

You are correct about the version. Forgot to increment it in `ext_emconf.php` - will fix.

**Comment by ghost** [comment]
Didn't do it manually; did a `composer require namelesscoder/inline-page-editing` and then activated extension in TYPO3 via Extension Manager. I have the require line in composer.json, but the autoload line wasn't added... why not?

**Comment by NamelessCoder** [comment]
I have only been able to reproduce this one on sites where the files were manually uploaded and `composer install` or `composer dump-autoload` was _not_ executed. I have checked this on a case insensitive file system as well, so it doesn't seem to be the class loader either. Are you sure this one is an issue? Could there be external causes why the class loading might fail (symlink structure, location of composer relative to project, changed vendor-dir, other...?)

> I have the require line in composer.json, but the autoload line wasn't added... why not?

It is only needed when you install the extension manually and it will never be added automatically, so that's as expected.
